### ISSUE TITLE
Pulseq importer does not support the rotation extension fully

### ISSUE THREAD

**Issue by J-Endres** [MRsources/MRzero-Core#182]
When using the pulseq rotation extension in a sequence block with gradients of different types, shapes or delays the import crashes.
This is because rotation will create newly sampled gradients that are a interpolation of multiple gradients of different types.

The error message will look like the following:
```
PanicException: not implemented: transform_grad: rotation across axes with different shapes or delays is not supported yet
```

This feature currently does not have high priority - the only example we could find which uses this feature was [writeFastRadialGradientEcho_rotExt.m at](https://github.com/pulseq/pulseq/blob/master/matlab/demoSeq/writeFastRadialGradientEcho_rotExt.m).
If you disagree and need that feature, please upvote this issue.

**Comment by WSHoge** [comment]
I'm happy to be the first to up-vote this feature request.  The Rotation Extension resolved a significant deficiency within Pulseq, and I'll be unable to simulate my radial sequences without this support in MRzero.
-- Scott Hoge

**Comment by YannBov1** [comment]
I also need this feature. It is not just useful for radial sequences but also to compress the file size of large sequences by re-using existing gradients.

For now I modify the mr0 sequence gradients after loading the sequence from file to obtain the rotated gradients:


```
from pypulseq import Sequence, calc_rf_center
import MRzeroCore as mr0
import numpy as np
import torch

def load_sequence(seq_file: str, pulseq_seq: Sequence, device: torch.device) -> mr0.Sequence:
    """Load a sequence and replace pulseq_rs gradient moments with PyPulseq's."""
    seq: mr0.Sequence = mr0.Sequence.import_file(
        seq_file, exact_trajectories=False, backend="pulseq_rs"
    )

    rf_times = []
    block_start = 0.0
    for block_id, block_duration in pulseq_seq.block_durations.items():
        block = pulseq_seq.get_block(block_id)
        if block.rf is not None:
            rf_center = float(calc_rf_center(block.rf)[0])
            rf_times.append(block_start + block.rf.delay + rf_center)
        block_start += block_duration
    rf_times = np.asarray(rf_times)
    if rf_times.size != len(seq):
        raise ValueError(
            f"PyPulseq found {rf_times.size} RF pulses but MRzero "
            f"created {len(seq)} repetitions"
        )

    gradient_moments = [
        None if gradient is None else gradient.antiderivative()
        for gradient in pulseq_seq.get_gradients(
            trajectory_delay=0, gradient_offset=0
        )
    ]
    for rep, rf_time in zip(seq, rf_times):
        event_durations = rep.event_time.detach().cpu().numpy().astype(float)
        event_boundaries = rf_time + np.concatenate(
            ([0.0], np.cumsum(event_durations))
        )
        corrected_gradm = np.zeros(tuple(rep.gradm.shape), dtype=np.float64)
        for axis, moment in enumerate(gradient_moments):
            if moment is None:
                continue

            # PPoly extrapolates outside its domain. Gradients are zero there,
            # so the antiderivative must instead remain constant.
            bounded_times = np.clip(event_boundaries, moment.x[0], moment.x[-1])
            corrected_gradm[:, axis] = np.diff(moment(bounded_times))

        rep.gradm.copy_(
            torch.as_tensor(
                corrected_gradm,
                dtype=rep.gradm.dtype,
                device=rep.gradm.device,
            )
        )

    if torch.device(device).type == "cuda":
        return seq.cuda()
    return seq
```
